tools/generator/landing-page.js

```diff
@@ -1,57 +1,235 @@
-
-// Features
-// TODO: If select "ungated" don't show show form template, campaign template, POI
-// TODO: Page generator will generate URL based on the content type and marquee title
-
-function withTimeout(promise, ms) {
-  return Promise.race([promise, new Promise((_, reject) => { setTimeout(() => reject(new Error('timeout')), ms); })]);
-}
+const AEM_PAGE = 'https://main--da-bacom--adobecom.aem.live';
+const DA_EDIT = 'https://da.live/edit#/adobecom/da-bacom';
+const PREVIEW_PARAMS = '?martech=off&dapreview=on';
+const FORM_STORAGE_KEY = 'landing-page-form-state';
+const OPTIONS_LOADING = [{ value: 'loading', label: 'Loading...' }];
+const OPTIONS_ERROR = [{ value: 'error', label: 'Error loading options' }];
+
+const TEMPLATE_MAP = {
+  guide: {
+    gated: '/tools/page-builder/prd-template-basic',
+    ungated: '/tools/page-builder/prd-template-basic-ungated',
+  },
+  report: {
+    gated: '/tools/page-builder/prd-template-basic',
+    ungated: '/tools/page-builder/prd-template-basic-ungated',
+  },
+  'video/demo': {
+    gated: '/tools/page-builder/prd-template-basic',
+    ungated: '/tools/page-builder/prd-template-video',
+  },
+  infographic: {
+    gated: '/tools/page-builder/prd-template-basic',
+    ungated: '/tools/page-builder/prd-template-basic-ungated',
+  },
+};
+
+const FORM_SCHEMA = {
+  contentType: '',
+  gated: '',
+  formTemplate: '',
+  campaignId: '',
+  marketoPOI: '',
+  marqueeEyebrow: '',
+  marqueeHeadline: '',
+  marqueeDescription: '',
+  marqueeImage: null,
+  bodyDescription: '',
+  cardTitle: '',
+  cardDescription: '',
+  contentTypeCaas: '',
+  primaryProduct: '',
+  seoMetadataTitle: '',
+  seoMetadataDescription: '',
+  experienceFragment: '',
+  videoAsset: '',
+  url: '',
+};
 
 class LandingPageForm extends LitElement {
-  static properties = {
-    data: { state: true },
-    marqueeImage: { state: true },
-  };
-
   static styles = style;
 
+  static get properties() {
+    return {
+      form: { type: Object },
+      marketoPOIOptions: { type: Array },
+      contentTypeOptions: { type: Array },
+      primaryProductOptions: { type: Array },
+      isInitialized: { type: Boolean },
+    };
+  }
+
   constructor() {
     super();
-    this.data = {};
-    this.marqueeImage = null;
+    this.form = { ...FORM_SCHEMA };
+    this.marketoPOIOptions = OPTIONS_LOADING;
+    this.contentTypeOptions = OPTIONS_LOADING;
+    this.primaryProductOptions = OPTIONS_LOADING;
+    this.isInitialized = false;
+    this.iframeRef = createRef();
+    this.template = null;
+    this.currentTemplateKey = null;
+  }
+
+  resetForm() {
+    this.form = { ...FORM_SCHEMA };
+    localStorage.removeItem(FORM_STORAGE_KEY);
+    if (this.isInitialized) {
+      this.requestUpdate();
+    }
+  }
+
+  saveFormState() {
+    const formToSave = { ...this.form };
+    // TODO: Handle other images
+    if (formToSave.marqueeImage && formToSave.marqueeImage.url && formToSave.marqueeImage.url.startsWith('blob:')) {
```

Comment:
What about shortening this with some optional chaining?
```suggestion
    if (formToSave.marqueeImage?.url?.startsWith('blob:')) {
```